youtube_dl/extractor/nexx.py

```diff
@@ -364,8 +364,19 @@ def _real_extract(self, url):
                     'X-Request-Token': request_token,
                 })
 
-        general = video['general']
-        title = general['title']
+        # some videos have 'bumpers' (ads), the API returns a list in that case
+        if type(video_json) == list:
+            for v in video_json:
+                general_info = v['general']
+                if str(general_info['ID']) == video_id:
+                    # the bumpers have differing IDs
+                    video = v
+                    general = general_info
+                    title = general['title']
```

Comment:
`next`.

Reply:
I don't understand what you're suggesting here, but I incorporated your other suggestions.